### PR TITLE
[master] ZIL-4874 revert block number '0' and fix getBlockByNumber tests

### PR DESCRIPTION
**Pull request by moboware**
## Backward Compatibility

- [x] This is not a breaking change
- [ ] This is a breaking change

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [ ] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

**Review by CSideSteve — COMMENTED** [review]
Could we not simply revert the commit on git ?

**Review by CSideSteve — APPROVED** [review]

File: tests/EvmAcceptanceTests/test/rpc/eth_getBlockByNumber.js [hunk 1]
```diff
@@ -127,49 +127,72 @@ describe("Calling " + METHOD, function () {
...
   });
 
-  it("should return get full transactions objects by 'earliest' tag", async function () {
-    await helper.callEthMethod(METHOD, 2, ["earliest", true], (result, status) => {
-      hre.logDebug(result);
+  describe("When on network", function () {
+    let blockNumber = 0x0;
+    before(async function () {
+      if (hre.network.name == "isolated_server") {
+        this.skip(); // isolated server does not returns block '0' back, but a null object.
```
Comment (its-saeed) [inline tests/EvmAcceptanceTests/test/rpc/eth_getBlockByNumber.js:134]: if this is a bug for isolated server, we can have a `FIXME` comment here with the related issue number included in.

File: tests/EvmAcceptanceTests/test/rpc/eth_getBlockByNumber.js [hunk 2]
```diff
@@ -127,49 +127,72 @@ describe("Calling " + METHOD, function () {
     });
   });
 
-  it("should return get full transactions objects by 'earliest' tag", async function () {
-    await helper.callEthMethod(METHOD, 2, ["earliest", true], (result, status) => {
-      hre.logDebug(result);
+  describe("When on network", function () {
```
Comment (its-saeed) [inline tests/EvmAcceptanceTests/test/rpc/eth_getBlockByNumber.js:130]: can you make this `describe` block description more specific?

**Review by its-saeed — APPROVED** [review]
two tiny notes

File: tests/EvmAcceptanceTests/test/rpc/eth_getBlockByNumber.js (hunk 1 above)
Comment (moboware) [reply tests/EvmAcceptanceTests/test/rpc/eth_getBlockByNumber.js:134]: i have a ticket for it

**Comment by moboware** [comment]
> 
We can revert the iso svr changes, but the test expects specific results and that breaks because they are different per network

**Pull request merged**